arviz/plots/forestplot.py

```diff
@@ -145,8 +145,9 @@ def plot_forest(
     """
     if not isinstance(data, (list, tuple)):
         data = [data]
+
     if transform is not None:
-        data = transform(data)
+        data = [transform(datum) for datum in data]
```

Comment:
This has to be moved below to 

```
datasets = [convert_to_dataset(datum) for datum in reversed(data)]
if transform is not None:
    datasets = [transform(dataset) for datum in datasets]
datasets = get_coords(datasets, list(reversed(coords)) if isinstance(coords, (list, tuple)) else coords)

```

data is still of type inference data, tests are catching that the issue was not yet solved, so tests are working perfectly!

Reply:
Ow, ok! Very clear, thanks Oriol!